src/decoder/mod.rs

```diff
@@ -1066,54 +1164,98 @@ impl<R: Read + Seek> Decoder<R> {
         }
     }
 
-    /// Read a single strip from the image and return it as a Vector
+    /// Read a single strip from the image and return it as a Vector. This method does not return
+    /// information on the size of the strip read and so the caller must take care of the final
+    /// strip in the image as the data read may be less than the expected strip size. This interface
+    /// may change at a future date to reflect this.
+    ///
+    /// `read_strip_at` can be used to get both the strip data and the strip size.
+    ///
+    #[deprecated = "Use read_chunk instead"]
     pub fn read_strip(&mut self) -> TiffResult<DecodingResult> {
-        self.check_chunk_type(ChunkType::Strip)?;
-        let index = self.current_chunk;
+        let result = self.read_strip_at(self.current_chunk)?;
 
-        let rows_per_strip =
-            usize::try_from(self.image().strip_decoder.as_ref().unwrap().rows_per_strip)?;
+        Ok(result.0)
+    }
 
-        let strip_height = cmp::min(
-            rows_per_strip,
-            usize::try_from(self.image().height)? - index * rows_per_strip,
-        );
+    /// Read the specified strip (at index `strip_index`) and return the binary data as a Vector and
+    /// `ChunkInfo` describing the size of the strip.
+    ///
+    #[deprecated = "Use read_chunk instead"]
+    pub fn read_strip_at(&mut self, strip_index: usize) -> TiffResult<(DecodingResult, ChunkInfo)> {
```

Comment:
Wait, you're adding this method and deprecating it at the same time?

Reply:
Ah I thought this was already in main...sorry it's been a while since I looked at the code!